src/pipes.cpp

```diff
@@ -0,0 +1,66 @@
+#include "pipes.h"
+
+const int Pipes::NUM_PIPES = 4;
+const float Pipes::PIPE_SCALE = 0.4;
+const int Pipes::PIPE_OFFSET_RANGE = 300;
+const float Pipes::SPEED_PER_SECOND = 500;
+
+Pipes::Pipes(int screenWidth, int screenHeight, int frameRate)
+{
+    bool res = pipeTexture.loadFromFile("./lib/pipe.png");
+    if(!res) throw("Cannot load pipe\n");
+
+    pipeSprites = new sf::Sprite[NUM_PIPES];
+    if (pipeSprites == nullptr) throw("Pipe array could not be created\n");
```

Comment:
C++ exceptions are not really common practice, especially in games.. Too many uncertainties about what might be going on. Handle your errors explicitly in this line rather than asking someone else to do it (with exceptions..)